fhir2dataset/parser.py

```diff
@@ -0,0 +1,193 @@
+import re
+from collections import defaultdict, OrderedDict
+
+
+PREFIX = [" eq ", " ne ", " gt ", " lt ", " ge ", " le ", " sa ", " eb ", " ap "]
+
+
+class FHIR2DatasetParser:
+    def __init__(self):
+        self.CLAUSES = OrderedDict()
+        self.__split_mask_clauses = ""
+
+        self.__init_clauses()
+        self.__init_split_mask()
+        self.__reset_config()
+
+    def parse(self, SQL_string: str):
+        self.__reset_config()
+        SQL_string = SQL_string.replace("\n", " ")
+        # print(f"SQL_string: {SQL_string}")
+
+        items = re.split(self.__split_mask_clauses, SQL_string)
+        items.remove("")
```

Comment:
Can you have `""` elements?

Reply:
Indeed this case occurs when the division element is at the beginning or at the end, for example, "SELECT p.given" becomes ["", "SELECT ", "p.given"].